paper/paper.md

```diff
@@ -0,0 +1,85 @@
+  - name: Henry Schreiner
+    orcid: 0000-0002-7833-783X
+    equal-contrib: true
+    affiliation: 1
+  - name: Jim Pivarski
+    orcid: 0000-0002-6649-343X
+    equal-contrib: true
+    corresponding: true
+    affiliation: 1
+  - name: Saransh Chopra
+    orcid: 0000-0003-3046-7675
+    equal-contrib: true
+    affiliation: 1
+
+affiliations:
+  - name: Princeton University
+    index: 1
+date: 16 August 2024
+bibliography: paper.bib
+---
+
+# Summary
+
+Vector algebra is a crucial component of data analysis pipelines in high energy
+physics, enabling physicists to transform raw data into meaningful results that
+can be visualized. Given that high energy physics data is not uniform, the
+vector algebra frameworks or libraries are expected to work readily on
+non-uniform or jagged data, allowing users to perform operations on an entire
+jagged array in minimum passes. Furthermore, optimizing memory usage and
+processing time has become essential with the increasing computational demands
+at the LHC. Vector is a Python library for creating and manipulating 2D, 3D,
+and Lorentz vectors, especially arrays of vectors, to solve common physics
+problems in a NumPy-like [@harris:2020] way. The library enables physicists to
+operate on high energy physics data in a high level language without
+compromising speed. The library is already in use at LHC and is a part of
+frameworks, like Coffea [@Gray:2023], employed by physicists across multiple
+high energy physics experiments.
+
+# Statement of need
+
+Vector is currently the only Lorentz vector library providing a Pythonic
+interface but a C++ (through Awkward Array [@Pivarski:2018]) computational
+backend. Vector integrates seamlessly with the existing high energy physics
+ecosystem and the broader scientific Python ecosystem, including libraries like
+Dask [@rocklin:2015] and Numba [@lam:2015]. The library implements a variety of
+backends for several purposes. Although vector was written with high energy
+physics in mind, it is a general-purpose library that can be used for any
+scientific or engineering application. The library houses 3+2 numerical
+backends for experimental physicists and 1 symbolic backend for theoretical
+physicists. These backends include a pure Python object backend for simple
+computations, a SymPy [@Meurer:2017] backend for symbolic computations, a
+NumPy backend for computations on regular data, an Awkward backend for
+computations on jagged data, and implementations of the Object and the Awkward
+backend in Numba for just-in-time compilable operations. Support for JAX and
+Dask is also provided through the Awkward backend, which enable vector
+functionalities to support automatic differentiation and parallel computing.
+
+## Impact
+
+Vector has become the de facto library for vector algebra in Python based high
+energy physics data analysis pipelines. The library has been installed over
+2 million times and 314 GitHub repositories use it as a dependency at the time
```

Comment:
Download count is a notoriously misleading metric of how often software is used. ("Notorious" because even though its problems are known, people still use it. It's hard to do anything better.) In particular,

* continuous testing frameworks and some highly parallel workloads will `pip install vector` or `conda install vector` as a first step, which inflates the numbers,
* it doesn't capture the difference between
  * users who download it once, never update versions, but use it every day and
  * (non-)users who update it daily with all the rest of the software on their computer, but never use it,
* it unintentionally captures the difference between
  * periods in which you release patches frequently (and users who live at head download every one of them),
  * periods in which there are no bugs so you don't release new versions at all (but users are still using it, all the same).

If you want to get quantitative in this paper, you might want to consider following the method of https://github.com/jpivarski-talks/2023-08-14-awkward-stats-update to

1. use GitHub's dependency graph to get a list of all the repos that _might_ be using Vector,
2. `git clone` them all,
3. search them for "import vector": `egrep -ral "(import\b.*\bvector|vector\b.*\bimport)" * --include="*.py" --include="*.ipynb"`,
4. use `git log --format=%cd "$z"` on all the matching files to find out when they were last touched,
5. make a plot like this:

<img src="https://github.com/jpivarski-talks/2023-08-14-awkward-stats-update/raw/main/PLOTS/appearances-of-import-awkward-cumulative-2024.svg" width="100%">

This would quantify the number of _direct_ users of Vector, the people who know that they are using it, as opposed to the people who get it through another library. Pretty soon, this won't be a good metric anymore because of indirect users through Coffea, but you could do a Vector + Coffea-vector plot then.

Another benefit of an analysis like this is that you can find out how users are using your interface—which functions they use most, and in what ways. I talked about that in [this presentation](https://indico.jlab.org/event/459/contributions/11547/) and I did a [similar analysis for Numba](https://github.com/jpivarski-talks/2024-02-13-numba-usage-stats).

Reply:
Thanks for the resources here! I went through them and also through several other JOSS papers. I did not statistics in any of them, so I think I will just remove them entirely.